load_test.py

```diff
@@ -37,7 +37,6 @@ def run_main_process(concurrency, duration, dataset, dataset_q, stop_q):
     while not dataset_q.empty():
         logging.debug("Removing element from dataset_q")
         dataset_q.get()
```

Comment:
I am still puzzled as to why it didn't remove all the elements from the dataset queue. 🤔

Reply:
It's because there is a background "QueueFeederThread" that is responsible for putting items in the Queue when the main process adds elements to it. However, Queue's aren't supposed to hold a long list of items so when the Queue gets too long the QueueFeederThread seems to hold some pending items until there is more space in the Queue, and then add them. So basically right after this loop, the background thread adds some more elements to the Queue, and then the main process won't exit because it is waiting for another process to take the items from the Queue.